src/main/java/io/hops/hopsworks/expat/migrations/featurestore/featureview/CreateFeatureViewFromTrainingDataset.java

```diff
@@ -0,0 +1,381 @@
+          "FROM training_dataset AS a " +
+          "INNER JOIN feature_store AS b ON a.feature_store_id = b.id " +
+          "INNER JOIN project AS c ON b.project_id = c.id " +
+          "INNER JOIN users AS d on a.creator = d.uid " +
+          "WHERE a.feature_view_id is null";
+  private final static String GET_ALL_FEATURE_VIEWS =
+      "SELECT fv.id, p.projectname, fv.version " +
+          "FROM feature_view AS fv " +
+          "INNER JOIN feature_store AS fs ON fv.feature_store_id = fs.id " +
+          "INNER JOIN project AS p ON fs.project_id = p.id";
+  private final static String GET_FEATURES =
+      "SELECT fg.name AS fg_name, fg.version AS fg_version, tdf.name AS name " +
+          "FROM training_dataset_feature AS tdf " +
+          "INNER JOIN feature_group AS fg ON tdf.feature_group = fg.id " +
+          "WHERE %s";
+
+  private final static String GET_ARRAY = "SELECT a.id FROM %s AS a WHERE %s";
+  private final static String SET_FEATURE_VIEW = "UPDATE %s SET feature_view_id = ? WHERE %s = ?";
+  private final static String REMOVE_FEATURE_VIEW_FROM_TABLES = "UPDATE %s SET %s = null WHERE %s = ?";
+  private final static String DELETE_FEATURE_VIEW = "DELETE FROM feature_view WHERE id = ?";
+
+  private final static String INSERT_FEATURE_VIEW =
+      "INSERT INTO feature_view(" +
+          // 1-5
+          "`name`, `feature_store_id`, `created`, `creator`, `version`, " +
+          // 6-10
+          "`description`, `label`, `inode_pid`, `inode_name`, `partition_id`" +
+          ") " +
+          "VALUE(" +
+          "?, ?, ?, ?, ?, " +
+          "?, ?, ?, ?, ?" +
+          ")";
+  private JAXBContext jaxbContext;
+
+  public CreateFeatureViewFromTrainingDataset() throws JAXBException {
+    super();
+    jaxbContext = jaxbContextMigrate();
+  }
+
+  public void runMigration() throws MigrationException {
+    // 1. Get all td
+    // 2. Create fv from td
+    // 3. set inode
+    // 4. set xattr
+    // 5. add fv to td, td feature, td join, td filter
+    try {
+      connection.setAutoCommit(false);
+      PreparedStatement getTrainingDatasetsStatement = connection.prepareStatement(GET_ALL_TRAINING_DATASETS);
+      PreparedStatement insertFeatureViewStatement =
+          connection.prepareStatement(INSERT_FEATURE_VIEW, Statement.RETURN_GENERATED_KEYS);
+      ResultSet trainingDatasets = getTrainingDatasetsStatement.executeQuery();
+      Integer n = 0;
+      while (trainingDatasets.next()) {
+        n += 1;
+        String name = trainingDatasets.getString("name");
+        Integer featurestoreId = trainingDatasets.getInt("feature_store_id");
+        Timestamp created = trainingDatasets.getTimestamp("created");
+        Integer creator = trainingDatasets.getInt("creator");
+        Integer version = trainingDatasets.getInt("version");
+        String description = trainingDatasets.getString("description");
+        //TODO: extract label
+        String label = "";
```

Comment:
What about this?

Reply:
This is not needed because label is stored as attribute of feature.